### PR TITLE
Added excludeAction param

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Adding excludeActions parameter to fileActions registerDefaultActions

## Motivation and Context
Imagine you write your own file list for a specific app, you want you to use all the default actions, but not the 'Rename' action.
Therefore you needed to define all the actions again in the app. This can now be overcome by passing ['Rename'] as excludeActions parameter. 
 

## How Has This Been Tested?

Manual UI Testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
@phil-davis any clue why the js tests are failing ?

File: apps/files/js/fileactions.js [hunk 1]
```diff
@@ -615,47 +615,85 @@
 
 		/**
 		 * Register the actions that are used by default for the files app.
+		 * @param {['Download', 'Rename', 'Delete']|null} [excludeAction] exclude actions
```
Comment (JammingBen) [inline apps/files/js/fileactions.js:618]: `excludeActions` (plural) seems more suitable.

File: apps/files/js/fileactions.js [hunk 2]
```diff
@@ -615,47 +615,85 @@
...
-				displayName: t('files', 'Rename'),
-				mime: 'all',
-				order: -30,
-				permissions: OC.PERMISSION_UPDATE,
-				iconClass: 'icon-rename',
-				actionHandler: function (filename, context) {
-					context.fileList.rename(filename);
+
+			defaultActions.forEach(function (action) {
+				if (!excludeAction.includes(action.name)) {
```
Comment (JammingBen) [inline apps/files/js/fileactions.js:695]: I believe we cannot use `includes()`, that's also probably why the tests fail. Better use `indexOf`, although it's ugly.

**Review by JammingBen — CHANGES_REQUESTED** [review]

**Comment by JammingBen** [comment]
> @phil-davis any clue why the js tests are failing ?

As mentioned in the review, I assume it's due to `includes()`. At least we can give it a try 🙂

**Review by JammingBen — COMMENTED** [review]
Nice! Add a changelog, then it's good to go from my side 🚀

**Review by JammingBen — APPROVED** [review]

**Comment by JanAckermann** [comment]
@jvillafanez cool with it?

**Comment by jvillafanez** [comment]
My only "complain" is that the `defaultActions` variable seems too big and it makes the function difficult to read. It's a bit easy to lose focus on what the function is doing. Taking into account that we don't plan to add more default actions there, it could be fine since we probably don't want to keep a copy of that variable somewhere else.
Anyway, I'm approving since I don't have better options at the moment.

**Review by jvillafanez — APPROVED** [review]

**Pull request merged**